### PR TITLE
Add flow run urls to notifications

### LINKED ISSUE #6805: Add `flow_run_url` to flow run notification template variables
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

Flow runs notifications currently allow the following fields to be provided as template variables

```
FLOW_RUN_NOTIFICATION_TEMPLATE_KWARGS = [
    "flow_run_notification_policy_id",
    "flow_id",
    "flow_name",
    "flow_run_url",
    "flow_run_id",
    "flow_run_name",
    "flow_run_parameters",
    "flow_run_state_type",
    "flow_run_state_name",
    "flow_run_state_timestamp",
    "flow_run_state_message",
]
```

### Describe the proposed behavior

Add `flow_run_url` to the allowed template variables.

The `FlowRunNotifications` service will be responsible for figuring out the correct flow run url and adding it to the templated variables.

### Example Use

If my flow run fails, I would like a slack notification with a link to the flow run ui page.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zangell44**
This PR adds `flow_run_url` as an available template variable for flow run notifications. We had this feature in v1 and it's a popular request for v2.

The default notification template now includes both the flow run url and state message.

Closes #6805 & https://github.com/PrefectHQ/prefect/issues/6604

### Example

Now you can create a Slack webhook with a message like

> Flow run `bad-example` now in state `Completed`. Read more detail here `<link to flow run>`.

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a `fix`, `feature`, `enhancement`, `docs`, `collections`, or `maintenance` label categorizing the change.

### CODE REVIEW

**Comment by zhen0** [comment]
Could we add the flow run url to the default message too?

**Comment by zangell44** [comment]
> Could we add the flow run url to the default message too?

Sure thing!

**Comment by zhen0** [comment]
I've tested this locally and looks good to me.  Could we get a code review and get this merged before the next release?

File: src/prefect/orion/models/flow_run_notification_policies.py [hunk 1]
```diff
@@ -19,6 +19,8 @@
 
     Flow ID: {flow_id}
     Flow run ID: {flow_run_id}
+    Flow run url: {flow_run_url}
```
Comment (madkinsz) [inline src/prefect/orion/models/flow_run_notification_policies.py:22]:
```suggestion
    Flow run URL: {flow_run_url}
```

**Pull request merged**